expyfun/stimuli/_texture.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""Texture (ERB-spaced) stimulus generation functions."""
+
+# adapted (with permission) from code by Hari Bharadwaj
+
+import numpy as np
+import warnings
+
+from ._stimuli import rms, window_edges
+
+
+def _cams(f):
+    """Compute cams."""
+    return 21.4 * np.log10(0.00437 * f + 1)
+
+
+def _inv_cams(E):
+    """Compute cams inverse."""
+    return (10 ** (E / 21.4) - 1.) / 0.00437
+
+
+def _scale_sound(x):
+    """Scale appropriately to between +/- 1."""
+    return 0.95 * x / np.max(np.abs(x))
+
+
+def _make_narrow_noise(bw, f_c, dur, fs, ramp_dur, rng):
+    """Make narrow-band noise using FFT."""
+    f_min, f_max = f_c - bw / 2., f_c + bw / 2.
+    t = np.arange(int(round(dur * fs))) / fs
+    # Make Noise
+    f_step = 1. / dur  # Frequency bin size
+    h_min = int(np.ceil(f_min / f_step))
+    h_max = int(np.floor(f_max / f_step)) + 1
+    phase = rng.rand(h_max - h_min) * 2 * np.pi
+    noise = np.zeros(len(t) // 2 + 1, np.complex)
+    noise[h_min:h_max] = np.exp(1j * phase)
+    return window_edges(np.fft.irfft(noise)[:len(t)], fs, ramp_dur,
+                        window='dpss')
+
+
+def texture_ERB(n_freqs=20, n_coh=None, rho=1., seq=('inc', 'nb', 'inc', 'nb'),
+                fs=24414.0625, dur=1., SAM_freq=7., random_state=None,
+                freq_lims=(200, 8000)):
+    """Create ERB texture stimulus
+
+    Parameters
+    ----------
+    n_freqs : int
+        Number of tones in mixture (default 20).
+    n_coh : int | None
+        Number of tones to be temporally coherent. Default (None) is
+        ``int(np.round(n_freqs * 0.8))``.
+    rho : float
+        Correlation between the envelopes of grouped tones (default is 1.0).
+    seq : list
+        Sequence of incoherent ('inc'), coherent noise envelope ('nb'), and
+        SAM ('sam') mixtures. Default is ``('inc', 'nb', 'inc', 'nb')``.
+    fs : float
+        Sampling rate in Hz.
+    dur : float
+        Duration (in seconds) of each token in seq (default is 1.0).
```

Comment:
might be nice for this to accept a sequence of durations, like if you wanted the `inc` to be 1.5 s followed by `nb` of 0.5 s.  But that can be a separate PR later, on YAGNI grounds.